pywisp/widgets/fileselector.py

```diff
@@ -14,23 +16,38 @@ def __init__(self, formats):
         super().__init__()
         self.st = settings(_default)
         self.formats = formats
+        self.exts = [self._extFromFilter(flt) for flt in formats]
+        self.logger = logging.getLogger("FileSelector")
+
+    def _extFromFilter(self, flt):
+        return re.search(r'\*([.a-z]*)', flt).group(1)
 
     def getSaveFileName(self, text="Export as ..."):
         path = self.st.value("path/export_dir")
         ext = self.st.value("path/export_ext")
-        file = os.path.join(path, "export" + ext)
+        file = os.path.join(path, "export")
```

Comment:
I would really like to keep the last exported file type as a suggestion for the extension. 
Ideally, the last file type format would also be used as first entry of 'filterStr'.

Reply:
The last used format is already the selected one when the FileSelector opens up.

Reply:
Sorry, I did not realize that. However, I still believe that being verbose about whats happening and putting the last used extension (which matches the last used filter as I now know) directly in the file name surely won't hurt.